generic_domains.php

```diff
@@ -142,7 +145,29 @@ public function addService(
      */
     public function editService($package, $service, array $vars = [], $parent_package = null, $parent_service = null)
     {
-        return $this->addService($package, $vars, $parent_package, $parent_service);
+        // Get current service fields
+        $service_fields = isset($service->fields) ? $this->serviceFieldsToObject($service->fields) : [];
```

Comment:
This should default to an object instead of an array.

`$service_fields = isset($service->fields) ? $this->serviceFieldsToObject($service->fields) : (object) [];`

Reply:
Updated